core/domain/question_services.py

```diff
@@ -85,31 +86,113 @@ def _create_new_question(committer_id, question, commit_message):
     create_question_summary(question.id, committer_id)
 
 
-def create_new_question_skill_link(question_id, skill_id, skill_difficulty):
-    """Creates a new QuestionSkillLink model.
+def create_multi_question_skill_links_for_question(
+        user_id, question_id, skill_ids, skill_difficulties):
+    """Creates multiple new QuestionSkillLink models. Also adds
+    the skill ids to the linked skill ids of the Question model.
 
     Args:
+        user_id: str. ID of the creator.
+        question_id: str. ID of the question linked to the skills.
+        skill_ids: list(str). ID of the skills to which the question is linked.
+        skill_difficulties: list(float). The difficulty of the question with
+            respect to a skill, represented by a float between
+            0 and 1 (inclusive).
+
+    Raises:
+        Exception. Skill difficulties and skill ids should match. The lengths of
+        the two lists are different.
+    """
+    if len(skill_ids) != len(skill_difficulties):
+        raise Exception(
+            'Skill difficulties and skill ids should match. The lengths of the '
+            'two lists are different.')
+    question = get_question_by_id(question_id)
+
+    new_question_skill_link_models = []
+    for index, skill_id in enumerate(skill_ids):
+        new_question_skill_link_models.append(
+            question_models.QuestionSkillLinkModel.create(
+                question_id, skill_id, skill_difficulties[index]))
+
+    new_linked_skill_ids = copy.deepcopy(question.linked_skill_ids)
+    new_linked_skill_ids.extend(skill_ids)
```

Comment:
Should we be checking: (a) the skilll_id is not in the list, and (b) the resulting list continues to contain less than MAX_SKILLS_PER_QUESTION skills? Should have backend tests for these, too.

(I think @sophiewu6 is suggesting making use of MAX_SKILLS_PER_QUESTION in your PR. I had also thought that a question is linked to at most 3 skill IDs, but I seem to remember that that was due to issues where we were trying to translate "fetch questions" into "fetch question skill links" in one of @sophiewu6's PRs,  and needed to keep a bound on things. If our query policy has changed now so that this bound is not needed, then it may not matter any more, but we should make sure. If we aren't sure yet -- I know there are several PRs in flux -- then I suggest imposing a bound for now, and lifting it later once everything has been thought through and is fully settled, and we're confident that we can handle larger numbers of associated skills.)

Reply:
It was a restriction in get_question_summaries_and_skill_descriptions in question_services.py, so when I was working on that PR I was assuming that. Could you provide some details about "our query policy has changed now so that this bound is not needed"? Does that mean the restriction doesn't apply anymore?

Reply:
Checking skill id is not in the list can become a quadratic time function. Rather extending it and then doing list(set()) to remove duplicates seems like a faster operation.

I am a bit confused of the purpose of MAX_SKILLS_PER_QUESTION, I do not see it being used in the PR that creates it?

Reply:
It's used in gae_models.py of https://github.com/oppia/oppia/pull/6927/files this PR. I was using that because I thought there is a restriction about at most three skills per question, and Sean also mentioned to add that check to QuestionCreationHandler to validate if there are too many skill_ids.

Just want to make sure if that restriction actually exists. If not, what I was doing to fetch questionSkillsLinks might be not safe because if users create a bunch of questions with more than three questions, the function in gae_models.py won't fetch enough links.

Reply:
Discussed this with @sophiewu6 , we are considering making this check in a separate PR
Reason being: Currently we do not enforce this check in any place where we add skills, we will have to enforce it everywhere. I don't want to mix those changes with the ones in this PR.